File: Web/eUniversity.Web/Scripts/viewModels/group/groupRowViewModel.js
function GroupRowViewModel(serverModel, save) {
    var self = this;
    self.Id = ko.observable();
    self.Name = ko.observable();
    self.Description = ko.observable();
    self.SpecializationId = ko.observable();
    self.DateOfAdmissionDisplay = ko.observable();

    self.IsSelected = ko.observable();

    if (serverModel != null)
        ko.mapping.fromJS(serverModel, {}, self);
    
    self.Save = function (viewModel) {
        viewModel.DateOfAdmissionDisplay(viewModel.DateOfAdmissionDisplay().format('YYYY-MM-DD'));
        
        save({ viewModels: [ko.mapping.toJS(viewModel)] });
    };
}

File: Business/eUniversity.Business.ViewModels/Group/GroupRowViewModel.cs
﻿using System;
using System.Globalization;
using eUniversity.Business.ViewModels.Base;
using eUniversity.Business.ViewModels.Enums;

namespace eUniversity.Business.ViewModels.Group
{
    /// <summary>
    /// View model for group row view model
    /// </summary>
    public class GroupRowViewModel : KnockoutViewModel, IViewModel
    {
        public long Id { get; set; }
        public string Name { get; set; }

        public string Description { get; set; }
        public string SpecializationId { get; set; }

        public string DateOfAdmissionDisplay { get; set; }

        public DateTime DateOfAdmission
        {
            get
            {
                if (string.IsNullOrEmpty(DateOfAdmissionDisplay))
                {
                    return DateTime.Now;
                }

                return DateTime.ParseExact(DateOfAdmissionDisplay, DateFormat, CultureInfo.InvariantCulture);
            }
            set
            {
                DateOfAdmissionDisplay = value.ToString(DateFormat);
            }
        }

        public string LastModified { get; set; }

        public EntityStatusEnum Status { get; set; }

        private const string DateFormat = "yyyy-MM-dd";
    }
}